File: 1047-Remove-Duplicates/1047.py
class Solution:
  def removeDuplicates(self, S: str) -> str:
    stk = list()
    for ch in S:
      if stk and stk[-1] == ch:
        stk.pop()
      else:
        stk.append(ch)
    return "".join(stk)

File: 1001-1100/1047-Remove-Duplicates/1047.java
class Solution {
  public String removeDuplicates(String S) {
    StringBuffer stack = new StringBuffer();
    int top = -1;
    for (int i = 0; i < S.length(); ++i) {
      char ch = S.charAt(i);
      if (top >= 0 && stack.charAt(top) == ch) {
        stack.deleteCharAt(top);
        --top;
      } else {
        stack.append(ch);
        ++top;
      }
    }
    return stack.toString();
  }
}